src/WindowsUtils/include/WindowsUtils/BusyLoopUtils.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WINDOWS_UTILS_BUSY_LOOP_UTILS_H_
+#define WINDOWS_UTILS_BUSY_LOOP_UTILS_H_
+
+// clang-format off
+#include <Windows.h>
+// clang-format on
+
+#include <stdint.h>
+#include <string>
+#include "OrbitBase/Result.h"
+
+namespace orbit_windows_utils {
+
+// Information returned from "InstallBusyLoopAtAddress" required to resume the original code.
+struct BusyLoopInfo {
+  uint32_t process_id = 0;
+  uint64_t address = 0;
+  // TODO (b/228234988) remove std::string from Write/ReadProcessMemory interface.
+  // We use std::string as a byte buffer here.
+  std::string original_bytes;
+};
+
+// Overwrite instructions at address by a 2 bytes busy loop.
+ErrorMessageOr<BusyLoopInfo> InstallBusyLoopAtAddress(HANDLE process_handle, void* address);
+// Replace the instructions overwritten by "InstallBusyLoopAtAddress" by the original instructions.
+ErrorMessageOr<void> RemoveBusyLoop(const BusyLoopInfo& busy_loop_info);
+// Set instruction pointer for given thread.
```

Comment:
This requires the thread to be suspended I suppose? Same for \*BusyLoop\* above?

Reply:
Added comments.

Reply:
InstallBusyLoopAtAddress doesn't need a thread to be suspended however. This can be called before a function is ever called.